File: CS/PopupContainerEditOKButton.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DevExpress.XtraEditors;
using System.ComponentModel;

namespace PopupContainerControlWithOKButton
{
    class PopupContainerEditOKButton : PopupContainerEdit
    {
        // static constructor
        static PopupContainerEditOKButton() { RepositoryItemPopupContainerEditOKButton.RegisterPopupContainerEditOKButton(); }

        // public constructor
        public PopupContainerEditOKButton() { }

        // ovverride property
        public override string EditorTypeName
        {
            get { return RepositoryItemPopupContainerEditOKButton.PopupContainerEditOKButtonEditorName; }
        }

        // property as corresponded repositoryitem
        [DesignerSerializationVisibility(DesignerSerializationVisibility.Content)]
        public new RepositoryItemPopupContainerEditOKButton Properties
        {
            get { return base.Properties as RepositoryItemPopupContainerEditOKButton; }
        }

        protected override DevExpress.XtraEditors.Popup.PopupBaseForm CreatePopupForm()
        {
            return new PopupContainerFormOkButton(this);
        }
    }
}

File: VB/PopupContainerEditOKButton.vb
Imports DevExpress.XtraEditors
Imports System.ComponentModel

Namespace PopupContainerControlWithOKButton

    Friend Class PopupContainerEditOKButton
        Inherits PopupContainerEdit

        ' static constructor
        Shared Sub New()
            RepositoryItemPopupContainerEditOKButton.RegisterPopupContainerEditOKButton()
        End Sub

        ' public constructor
        Public Sub New()
        End Sub

        ' ovverride property
        Public Overrides ReadOnly Property EditorTypeName As String
            Get
                Return RepositoryItemPopupContainerEditOKButton.PopupContainerEditOKButtonEditorName
            End Get
        End Property

        ' property as corresponded repositoryitem
        <DesignerSerializationVisibility(DesignerSerializationVisibility.Content)>
        Public Overloads ReadOnly Property Properties As RepositoryItemPopupContainerEditOKButton
            Get
                Return TryCast(MyBase.Properties, RepositoryItemPopupContainerEditOKButton)
            End Get
        End Property

        Protected Overrides Function CreatePopupForm() As Popup.PopupBaseForm
            Return New PopupContainerFormOkButton(Me)
        End Function
    End Class
End Namespace
